sklearn/utils/estimator_checks.py

```diff
@@ -355,10 +357,38 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def gram_matrix_if_pairwise(X, estimator, kernel=linear_kernel):
+
+    if len(X.shape) == 1:
+        X = X.reshape(-1, 1)
+
+    if is_pairwise_metric(estimator):
+        # pairwise_distance() fails for certain versions of scipy
```

Comment:
`pairwise_distances` in scikit-learn fails for supported versions of scipy? can't be possible. you might have just got a random CI failure??

Reply:
That was my thinking. For certain versions I'd get an error saying I was `pairwise_distances()` passing a 0-dimensional matrix. Only happened for the Travis tests.

Reply:
@jnothman should I go back and use the scipy method even if it fails some of the CI tests? It tells me that I'm passing a 0-dimensional array which is clearly a bug

Reply:
I'd like to check the full error traceback to try understand what's going on, thanks.